### PR TITLE
TYP: added missing __init__ in pandas/core/methods

### PR DESCRIPTION
**Pull request by twoertwein**
See https://github.com/microsoft/pyright/issues/4544#issuecomment-1407513778

Ruff's INP001 could have caught this. Could enable it if the following are ignored:

> ci/fix_wheels.py:1:1: INP001 File `ci/fix_wheels.py` is part of an implicit namespace package. Add an `__init__.py`.
> ci/test_wheels.py:1:1: INP001 File `ci/test_wheels.py` is part of an implicit namespace package. Add an `__init__.py`.
> doc/make.py:1:1: INP001 File `doc/make.py` is part of an implicit namespace package. Add an `__init__.py`.
> doc/scripts/eval_performance.py:1:1: INP001 File `doc/scripts/eval_performance.py` is part of an implicit namespace package. Add an `__init__.py`.
> doc/source/conf.py:1:1: INP001 File `doc/source/conf.py` is part of an implicit namespace package. Add an `__init__.py`.
> setup.py:1:1: INP001 File `setup.py` is part of an implicit namespace package. Add an `__init__.py`.
> typings/numba.pyi:1:1: INP001 File `typings/numba.pyi` is part of an implicit namespace package. Add an `__init__.py`.
> web/pandas_web.py:1:1: INP001 File `web/pandas_web.py` is part of an implicit namespace package. Add an `__init__.py`.

### CODE REVIEW

**Comment by twoertwein** [comment]
There are some new errors detected - will try to address or ignore them.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -135,7 +135,7 @@ repos:
         types: [python]
         stages: [manual]
         additional_dependencies: &pyright_dependencies
-        - pyright@1.1.276
+        - pyright@1.1.284
```
Comment (twoertwein) [inline .pre-commit-config.yaml:138]: There are more errors with 1.1.291 https://github.com/microsoft/pyright/issues/4546

**Comment by twoertwein** [comment]
The second commit fixes type issues uncovered by a newer pyright version. The third commit uses Sequence[Hashable] more consistently.

**Comment by mroeschke** [comment]
Looks good just a merge conflict

**Comment by twoertwein** [comment]
> Looks good just a merge conflict

Rebased and green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**